employees/forms.py

```diff
@@ -89,41 +89,35 @@ def __init__(self, *args: Any, **kwargs: Any):
         self.fields["project"].queryset = (
             Project.objects.filter_active().filter(Q(members=author) | Q(managers=author)).distinct()
         )
-        self._filter_task_activities_per_project()
-        if "data" not in kwargs:
-            self._set_last_choices_in_report_form(author)
-
-    def _filter_task_activities_per_project(self) -> None:
-        if "project" in self.data:
-            with suppress(ValueError, TypeError):
-                project_id = int(self.data.get("project"))
-                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(projects=project_id).order_by(
-                    "name"
-                )
-        elif self.instance.pk:
-            self.fields["task_activities"].queryset = (
-                self.instance.project.project_activities.order_by("name")
-                | TaskActivityType.objects.filter(pk=self.instance.task_activities.pk)
-            ).distinct()
+        self._filter_task_activities(author)
 
-    def _set_last_choices_in_report_form(self, author: CustomUser) -> None:
+    def _filter_task_activities(self, author: CustomUser) -> None:
```

Comment:
This method is not only filtering task activities so name is no proper for what this method is doing. Please do as I wrote regarding this whole method - split it for smaller well named methods and then you see what exactly is done in this method.

Reply:
changed